### PR TITLE
Paella 7 custom themes

### PR DESCRIPTION
**Pull request by miesgre**
This PR adds the new paella feature (skins) to Opencast.
With this, each tenant can configure the paella skin/theme (use other colors, or other icon set)

### Your pull request should…

* [x] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

**Comment by gregorydlogan** [comment]
Should this go into `r/14.x`?

**Comment by miesgre** [comment]
It would be great. I can redirect this PR to r/14.x.
What do you think @mliradelc ?

**Comment by mliradelc** [comment]
I agree with Greg! This should go to 14.x

**Comment by miesgre** [comment]
> I agree with Greg! This should go to 14.x

Done!

File: etc/ui-config/mh_default_org/paella7/default_theme/theme.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "styleSheets": [
```
Comment (karendolan) [inline etc/ui-config/mh_default_org/paella7/default_theme/theme.json]: The Opencast UIConfigRest.java has a GET  @Path("{component}/{filename}")
That service can't serve a file at @Path("{component}/{theme}/{filename}"). Should all these paella7 themes be copied into the engage-player-paella-7 or some other static hosting module prior to build via maven?

Comment (miesgre) [reply etc/ui-config/mh_default_org/paella7/default_theme/theme.json]: Thanks @karendolan for testing this. 
You are right. I tested it with the proxy and did not detect this problem. I thought that everything in that folder was 
served. Sorry.

I will try to fix it!

Comment (miesgre) [reply etc/ui-config/mh_default_org/paella7/default_theme/theme.json]: @karendolan I updated the PR.

**Comment by miesgre** [comment]
@mliradelc, @karendolan Can you review it?

File: docs/guides/admin/docs/modules/paella.player7/configuration.md [hunk 2]
```diff
@@ -28,10 +28,58 @@ The configurations for the player are done for each tenant. So the configuration
...
+    ...
+    "opencast": {
+        "theme": "default_teme"
+        ...
+    }
+    ...
+}
+```
+
+Default theme: `.../etc/ui-config/<tenant>/paella7/default_theme/theme.json`.
```
Comment (mliradelc) [inline docs/guides/admin/docs/modules/paella.player7/configuration.md]: There is no "default_theme" folder.

File: etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    "styleSheets": [
+        "theme.css"
```
Comment (mliradelc) [inline etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json:3]: There is a folder called `custom_theme_example` but also (Maybe left from another PR) `custom_theme.css`

**Review by mliradelc — CHANGES_REQUESTED** [review]
I tried the skin PR, and I don't know why, but I changed several colors from the theme, and nothing changed. I tried with `custom_theme_example/theme.css` and `custom_theme.css`, but no one affects the colors from the player.

I tried also disabling the browser cache without avail.

**Comment by stefanosgeo** [comment]
@miesgre  do you have time today to check for @mliradelc  comments? We are going to release Opencast 14.x in about 6 hours.

**Comment by miesgre** [comment]
> I tried the skin PR, and I don't know why, but I changed several colors from the theme, and nothing changed. I tried with `custom_theme_example/theme.css` and `custom_theme.css`, but no one affects the colors from the player.
> 
> I tried also disabling the browser cache without avail.


@stefanosgeo, @mliradelc, I updated the documentation.

To test it you need a to create a `custom theme`. As an example there is a `basic theme` called `custom_theme_example`.

Rename the folder to `custom_example` or change the config.json file the `opencast.theme` property to the folder name.
You can change the colors or icons to test it.

File: docs/guides/admin/docs/modules/paella.player7/configuration.md (hunk 2 above)
Comment (miesgre) [reply docs/guides/admin/docs/modules/paella.player7/configuration.md]: Documentation updated!

File: etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json (hunk 3 above)
Comment (miesgre) [reply etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json:3]: `custom_theme_example` is just an example. To use it you need to change in the config.json file the `opencast.theme` property to `custom_theme_example`.

`

**Comment by mliradelc** [comment]
Thanks for the changes.

I'm Testing right now :)

**Comment by miesgre** [comment]
@mliradelc You can find the `default opencast theme` here: https://github.com/opencast/opencast/tree/9294dbb6cf966695d5e4cb7a9dbb406347b83a20/modules/engage-paella-player-7/public/default_theme
but it is embedded in the paella7 bundle. It is used when no custom theme is defined or there is an error loading the resources.

File: etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json (hunk 3 above)
Comment (mliradelc) [reply etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json:3]: Now I checked between the compiled binary and the source code. In the binary, there is an additional file
 `custom_theme.css` that is outside the folder.
 
image

If the custom themes should be only in folders, what is that `CSS` file?

**Comment by mliradelc** [comment]
> @mliradelc You can find the `default opencast theme` here: https://github.com/opencast/opencast/tree/9294dbb6cf966695d5e4cb7a9dbb406347b83a20/modules/engage-paella-player-7/public/default_theme but it is embedded in the paella7 bundle. It is used when no custom theme is defined or there is an error loading the resources.

I supposed that the default theme was in the etc files, but is inside the jar bundle!

File: etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json (hunk 3 above)
Comment (miesgre) [reply etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json:3]: An old file. make a clean. That file should not be there.

Comment (mliradelc) [reply etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json:3]: Strange, I did a `mvn  clean install.` I will reset the folder and then a new build

**Review by mliradelc — APPROVED** [review]
Now I tested, and the themes changed. I can change the icons and some CSS colors. The browser loads the `CSS` file.

The additional `CSS` file doesn't appear anymore. That surprised me because a c"lean install"  was not enough.

One more thing,
I could not change the playback bar's color and couldn't find the CSS parameter to modify the color for that zone. 

I approve it because it works as expected.

File: etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json (hunk 3 above)
Comment (mliradelc) [reply etc/ui-config/mh_default_org/paella7/custom_theme_example/theme.json:3]: > An old file. make a clean. That file should not be there.

That was the problem `git clean -dfx` solved the issue.

**Pull request merged**